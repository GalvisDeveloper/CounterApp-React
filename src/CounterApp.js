// import moduleName from 'module';

import PropTypes from 'prop-types'
import React, { useState } from 'react';

const CounterApp = ({ value }) => {

    const [lamarca, setLamarca] = useState(value);

    // Funcion handleAdd
    // const handleAdd = () => {
    //     setLamarca(lamarca + 1);
    //     // Otra manera de hacerlo ===> setLamarca((m) => m+1);
    // }

    // Funcion handleReset
    // const handleReset = () => { setLamarca(value); }

    // Funcion handleSubstract
    // const handleSubstract = () => { setLamarca(lamarca - 1); }

    return (
        <>
            <h1>La Marca</h1>
            <h2>{lamarca}</h2>
            {/* <button onClick={handleAdd}> Add </button>
            <button onClick={handleReset}> Reset </button>
            <button onClick={handleSubstract}> Substract </button> */}

            {/* Sin tener que declarar las funciones por aparte */}
            <button onClick={(e) => { setLamarca(lamarca + 1) }} > Add </button>
            <button onClick={(e) => { setLamarca(value) }} > Reset </button>
            <button onClick={(e) => { setLamarca(lamarca - 1) }} > Substract </button>
        </>
    );

}

CounterApp.propTypes = {
    value: PropTypes.number.isRequired
}

export default CounterApp;