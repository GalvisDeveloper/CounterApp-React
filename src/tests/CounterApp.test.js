

// import '@testing-library/jest-dom';
import React from 'react';
import CounterApp from '../CounterApp';
import { shallow } from 'enzyme';

describe('Probando CounterApp', () => {

    let valor;
    let wrapper = shallow(<CounterApp value={0} />);

    beforeEach(() => {
        valor = 100;
        wrapper = shallow(<CounterApp value={valor} />);
    });

    test('Debe de mostrar <CounterApp /> Correctamente', () => {

        expect(wrapper).toMatchSnapshot();

    });


    test('Debe de mostrar el valor por defecto de 100 ', () => {

        const valorH2 = wrapper.find('h2').text();
        expect(parseInt(valorH2)).toBe(valor);

    });


    test('Probando handleSubstract ', () => {

        wrapper.find('button').at(2).simulate('click');

        const valorH2 = wrapper.find('h2').text().trim();

        expect(parseInt(valorH2)).toBe(valor - 1);

    });

    test('Probando handleAdd ', () => {

        wrapper.find('button').at(0).simulate('click');

        const valorH2 = wrapper.find('h2').text().trim();

        expect(parseInt(valorH2)).toBe(valor + 1);

    });

    test('Probando handleReset ', () => {

        wrapper.find('button').at(1).simulate('click');

        const valorH2 = wrapper.find('h2').text().trim();

        expect(parseInt(valorH2)).toBe(valor);

    });



});
