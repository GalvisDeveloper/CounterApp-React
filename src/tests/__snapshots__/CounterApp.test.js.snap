// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando CounterApp Debe de mostrar <CounterApp /> Correctamente 1`] = `
<Fragment>
  <h1>
    La Marca
  </h1>
  <h2>
    100
  </h2>
  <button
    onClick={[Function]}
  >
     Add 
  </button>
  <button
    onClick={[Function]}
  >
     Reset 
  </button>
  <button
    onClick={[Function]}
  >
     Substract 
  </button>
</Fragment>
`;
